lib/moonshell.sh

```diff
@@ -120,6 +97,23 @@ _moonshell_test () {
     popd >/dev/null
 }
 
+#
+# MOONSHELL SYSTEM FUNCTIONS
+#
+
+_moonshell_system_check () {
+    local moonshell_dir=$1
+
+    # As this function is only used when called from _moonshell_check we are
+    # making the small assumption that we have passed the root/sudo test, so
+    # this block should JustWork.
+    [[ ! $(whoami) =~ ^root$ ]] && local sudo=sudo
```

Comment:
Minor nit, but probably should define an empty local sudo above and just set it here, in case $sudo is set in a caller (or globally).